### DISCUSSION TITLE
Custom item fileds in grid compoenents

### DISCUSSION THREAD

**Discussion by cubecleveland** [react-grid-layout/react-grid-layout#1497] [General]
Hello,
How we set custom fields in grid components to stor the component type for example.

i tries adding a key to the props and it doesnt come back on Layout changed

My keys are defined like this:

const [state, setState] = React.useState({
    items: [0, 1, 2, 3, 4].map(function (i, key, list) {
      return {
        i: i.toString(),
        x: i * 3,
        y: 0,
        w: 3,
        h: 2,
      
        minW: 6,
        maxW: 4,
        settings: { a: "4" },
      };
    }),
    newCounter: 0,
  });

the key settings doesnt come back on layout change.
Is there any way to stor specific data about each grid iem that will be returned ?
Thanks!

**Comment by JBartscher** [comment]
I also fiddled with the idea of using a custom layout type to add custom properties to a layout respectively the layout items.
I defined a custom type that extends the `GridLayout.Layout` type which worked with all the handlers for dragging and instanitating the ResponsiveGridLayout.


```
/**
 * extends the common GridLayout.Layout type with an widgetType
 */
export interface WidgetLayoutType extends Layout {
    type : 'table' | 'bar' | 'pie'
}
```

As @cubecleveland already mentioned the problem is that the `onLayoutChange` handler always returns a layout of the original `GridLayout.Layout` type and any extensions of the type are lost.
